frontend/src/lib/components/InsightLegend/InsightLegend.tsx

```diff
@@ -67,6 +67,7 @@ export function InsightLegend(): JSX.Element {
                                             item.breakdown_value === '' ? 'None' : item.breakdown_value?.toString()
                                         }
                                         compareValue={filters.compare ? formatCompareLabel(item) : undefined}
+                                        pillMidEllipsis={item?.filter?.breakdown === '$current_url'}
```

Comment:
Curious why we only do this for `$current_url`, in any case adding a code comment could be useful.

Reply:
It would be more robust if we could define a set of breakdown values that would benefit from a mid ellipsis truncation. For now, I thought $current_url was the biggest offender of taking up too much space but not having differentiating enough text in the beginning or end of the string. But I agree I'll add a todo comment here for now.